yarn-project/end-to-end/src/e2e_card_game.test.ts

```diff
@@ -0,0 +1,297 @@
+import { AztecNodeService } from '@aztec/aztec-node';
+import { AztecRPCServer } from '@aztec/aztec-rpc';
+import { AztecAddress, Wallet } from '@aztec/aztec.js';
+import { DebugLogger } from '@aztec/foundation/log';
+import { CardGameContract } from '@aztec/noir-contracts/types';
+import { AztecRPC, CompleteAddress } from '@aztec/types';
+
+import { setup } from './fixtures/utils.js';
+
+/* eslint-disable camelcase */
+
+interface Card {
+  points: bigint;
+  strength: bigint;
+}
+
+const cardToField = (card: Card): bigint => {
+  return card.strength + card.points * 65536n;
```

Comment:
side q:
longer term, is this something that should be handled by a generated typescript class?  

also i think there is prover system abstraction leaking through (modulus/overflow is proving system dependent?)

Reply:
Nah this is only because our macros dont' support arrays of structs yet as arguments to contract functions, when that's updated we don't need to pack the card into a field, we can just pass a card struct

What do you mean about the modulus/overflow? the fact that we can pack the card (32 bits) into a single field? For aztec we can safely assume that we'll always have bn254 I think 🤔

Reply:
ahh OK got it!

ohh right for modulus I forgot aztec has the pinned backend vs "regular" noir.  i was thinking if points got really big we could maybe have overflow (in some setting besides the test)